distributed/protocol/cupy.py

```diff
@@ -3,6 +3,18 @@
 """
 import cupy
 from .cuda import cuda_serialize, cuda_deserialize
+from distutils.version import LooseVersion
+
+
+class PatchedCudaArrayInterface(object):
+    # TODO: This class wont be necessary
+    #       once Cupy<7.0 is no longer supported
+    def __init__(self, ary):
+        vsn = LooseVersion(cupy.__version__)
+        cai = ary.__cuda_array_interface__
+        if vsn < "7.0.0rc1" and cai.get("strides") is None:
```

Comment:
Should this concern us?

```python
In [1]: "7.0.0" < "7.0.0rc1"
Out[1]: True
```

Reply:
Argh, I'd say so, we want "7.0.0" inclusive. What is normally used for the proper comparison in such cases? In the dask codebase I see things like the following:

```python
PANDAS_GT_0240 = PANDAS_VERSION >= LooseVersion("0.24.0rc1")
```

That seems to be the same kind of comparison, and I'm assuming that works?

Reply:
Perhaps the newest revision is better?  I am creating an empty cupy object to check the *supported* `__cuda_array_interface__` version.  E.g:

```python
class PatchedCudaArrayInterface(object):
    # TODO: This class wont be necessary
    #       once Cupy<7.0 is no longer supported
    def __init__(self, ary):
        cai = ary.__cuda_array_interface__
        cai_cupy_vsn = cupy.ndarray(0).__cuda_array_interface__["version"]
        if cai.get("strides") is None and cai_cupy_vsn < 2:
            cai.pop("strides", None)
        self.__cuda_array_interface__ = cai
```

Reply:
That's a good idea indeed, thanks for coming up with this @rjzamora. I left a comment on improving this (or rather reverting or your latest change).